### PR TITLE
fix redirector unit file

### PR DESCRIPTION
**Pull request by heronhaye**
@strib tagging you, but of course this can wait until after you're back from vacation. I think this should get in for the release. Not sure why it only started breaking now, seems to have been incorrect all along.

### CODE REVIEW

**Comment by heronhaye** [comment]
@strib, ptal?

File: packaging/linux/post_install.sh [hunk 1]
```diff
@@ -166,6 +166,15 @@ fix_bad_config_perms() {
...
 }
 
+fix_bad_config_perms
+
+# Update the GTK icon cache, if possible.
+if command -v gtk-update-icon-cache &> /dev/null ; then
+  gtk-update-icon-cache -q -t -f /usr/share/icons/hicolor
+fi
+
+safe_restart_systemd_services # restart services before restarting redirector so any changes to redirector unit file will be picked up
```
Comment (strib) [inline packaging/linux/post_install.sh]: Put the comment above this line and format so it doesn't violate 80 char width?

File: packaging/linux/systemd/keybase-redirector.service [hunk 2]
```diff
@@ -12,6 +12,7 @@ EnvironmentFile=-%h/.config/keybase/keybase.env
 ExecStartPre=/usr/bin/keybase --use-root-config-file config get --direct --assert-false --assert-ok-on-nil disable-root-redirector
 ExecStart=/usr/bin/keybase-redirector /keybase
 Restart=on-failure
+KillMode=process
```
Comment (strib) [inline packaging/linux/systemd/keybase-redirector.service]: Please put a comment here for why this is needed, so it doesn't get blindly copied,

**Comment by heronhaye** [comment]
@strib I changed the strategy of this PR as per our discussion.

**Comment by strib** [comment]
Looks good, thanks!

**Pull request merged**